statistics-production/api-data-standards.qmd

```diff
@@ -0,0 +1,262 @@
+
+:::
+
+### Example 3 - Metrics with non-compatible filters
+
+#### Example of bad practice
+
+The following would not be accepted for publication via the API.
+
+::: {.table-responsive}
+
+| pupil_count_grade9to5 | pupil_count_grade9to4 | pupil_count_grade9to1 | pupil_percent_grade9to5 | pupil_percent_grade9to4 | pupil_percent_grade9to1 | progress8_score_male | progress8_score_female | progress8_score |attainment8_score_male | attainment8_score_female | attainment8_score |
+|------------------------|------------------------|------------------------|--------------------------|--------------------------|-------------------------|----------------------|------------------------|-----------------|----------------------|------------------------|-----------------|
+| 30                     | 40                     | 50                     | 60                       | 80                       | 100                      | 0.2                  | 0.21                   | 0.21            |0.09                  | 0.08                   | 0.10            |
+
+: Attainment grade rates and scores in non-tidy format
+
+:::
+
+In this case, the different metrics contain different types of values that are split by very different filters. Specifically pupil counts and pupil percents are split into grade thresholds, whereas the score based metrics are not. If we were to try and pivot this data as one file, it would lead to an unreasonably large number of cells with no valid entries (i.e. large numbers of z's). For example, pivoting might create something like the following table, which suffers from both a large number of not applicable columns and duplication of data unecessarily.
+
+::: {.table-responsive}
+
+| sex    | grade_range        |accountability_measure | pupil_count  | pupil_percent  | score_average |
+|--------|--------------------|-----------------------|--------------|----------------|---------------|
+| Total  | Grades 9-5         | z                     | 30           | 60             | z             |
+| Total  | Grades 9-4         | z                     | 40           | 80             | z             |
+| Total  | Grades 9-1         | z                     | 50           | 100            | z             |
+| Total  | z                  | Attainment 8          | 50           | 100            | 0.21          |
+| Female | z                  | Attainment 8          | 50           | 100            | 0.21          |
+| Male   | z                  | Attainment 8          | 50           | 100            | 0.20          |
+| Total  | z                  | Progress 8            | 50           | 100            | 0.08          |
+| Female | z                  | Progress 8            | 50           | 100            | 0.08          |
+| Male   | z                  | Progress 8            | 50           | 100            | 0.09          |
+
+: Example of pivoted data showing excessive duplicated and not applicable fields.
+
+:::
+
+::: {.table-responsive}
```

Comment:
Think this is a duplicate table you can drop?

Reply:
True!